core/modules/layout/layout.api.php

```diff
@@ -10,34 +10,66 @@
  *   - path: A local path within the providing module to files needed by this
- *     layout, such as associated CSS, the icon image, and template file.
- *   - regions: A list of regions this layout provides, keyed by a machine name
- *     with a human label value.
- *   - preview: Optional. An image representing the appearance of this layout.
- *     If left empty, "preview.png" will be used.
- *   - stylesheets: An array of CSS file used whenever this layout is presented.
- *     If left empty, "one-column.css" will be used for all media types.
+ *     layout template, such as associated CSS, the icon image, and tpl.php
+ *     file.
+ *   - regions: A list of regions this layout template provides, keyed by a
+ *     machine name with a human label value.
+ *   - preview: Optional. An image representing the appearance of this layout
+ *     template. If left empty, "preview.png" will be used.
+ *   - stylesheets: An array of CSS file used whenever this layout template
+ *     is presented. If left empty, "one-column.css" will be used for all media
+ *     types.
  *   - template: The name of the template file (without the extension) used for
- *     this layout. All layouts should always be named with a "layout--" prefix,
- *     so that the default variables may be provided in
+ *     this layout template. Layout templates should always be named with a 
+ *     "layout--" prefix, so that the default variables may be provided in
  *     template_preprocess_layout(). If left empty, "layout--[key]" will be
- *     used, with underscores converted to hyphens in the layout key.
- *   - file: The name of a PHP file to be included prior to any rendering of
- *     this layout. This may be used to provide preprocess functions to prepare
- *     variables for the use of the layout.
+ *     used, with underscores converted to hyphens in the layout template key.
+ *   - file: Optional. The name of a PHP file to be included prior to any
+ *     rendering of this layout template. This may be used to provide preprocess
+ *     functions to prepare variables for the use of the layout template.
  */
+function hook_layout_template_info() {
+  $layout_templates['my_layout_template'] = array(
+    'title' => t('A custom layout template'),
+    'path' => 'layouts/my-layout-template',
+    'regions' => array(
+      'header' => t('Header'),
+      'content' => t('Content'),
+      'sidebar' => t('Sidebar'),
+      'footer' => t('Footer'),
+    ),
+
+    // Optional information that populates using defaults.
+    // 'preview' => 'preview.png',
+    // 'stylesheets' => array('all' => array('one-column.css')),
+    // 'template' => 'layout--my-layout-template',
+
+    // Specify a file containing preprocess functions if needed.
+    // 'file' => 'my_layout_template.php',
+  );
+  return $layout_templates;
+}
+
+  /**
+   * Deprecated. Now replaced by hook_layout_template_info().
+   *
+   * @deprecated since 1.29.0
+   */
```

Comment:
Should we replace the contents of this with a "Don't use this - use hook_layout_template_info() instead" inline comment? (to discourage use)

Reply:
I considered this, but though deprecated its still a valid hook until 2.0 so we'd want the example to show in the docs site.